plugins/wazuh-core/public/services/http/server-client.ts

```diff
@@ -0,0 +1,513 @@
+    const overwriteHeaders =
+      typeof extraOptions.overwriteHeaders === 'object'
+        ? extraOptions.overwriteHeaders
+        : {};
+
+    try {
+      if (!method || !path) {
+        throw new Error('Missing parameters');
+      }
+
+      const timeout = await this.services.getTimeout();
+      const url = this.services.getURL(path);
+      const options = {
+        method: method,
+        headers: {
+          ...pluginPlatformRequestHeaders,
+          'content-type': 'application/json',
+          ...overwriteHeaders,
+        },
+        url: url,
+        data: payload,
+        timeout: timeout,
+      };
+      const data = await this.services.request(options);
+
+      if (data['error']) {
+        throw new Error(data['error']);
+      }
+
+      return data;
+    } catch (error) {
+      // if the requests fails, we need to check if the API is down
+      if (checkCurrentApiIsUp) {
+        const currentApi = this.services.getServerAPI();
+
+        if (currentApi) {
+          // eslint-disable-next-line no-useless-catch
+          try {
+            await this.checkAPIById(currentApi);
+          } catch (error) {
+            // TODO :implement
+            // const wzMisc = new WzMisc();
+            // wzMisc.setApiIsDown(true);
+            // if (
+            //   !NavigationService.getInstance()
+            //     .getPathname()
+            //     .startsWith('/settings')
+            // ) {
+            //   NavigationService.getInstance().navigate('/health-check');
+            // }
+            throw error;
+          }
+        }
+      }
+
+      // if(this.onErrorInterceptor){
+      //   await this.onErrorInterceptor(error, {checkCurrentApiIsUp, shouldRetry, overwriteHeaders})
+      // }
+      const errorMessage = error?.response?.data?.message || error?.message;
+
+      if (
+        typeof errorMessage === 'string' &&
+        errorMessage.includes('status code 401') &&
+        shouldRetry
+      ) {
+        try {
+          await this.auth(true); // await WzAuthentication.refresh(true);
+
+          return this.requestInternal(method, path, payload, {
+            shouldRetry: false,
+          });
+        } catch (error) {
+          throw this.returnErrorInstance(
+            error,
+            error?.data?.message || error.message,
+          );
+        }
+      }
+
+      throw this.returnErrorInstance(
+        error,
+        errorMessage || 'Server did not respond',
+      );
+    }
+  }
+
+  /**
+   * Perform a request to the Wazuh API
+   * @param {String} method Eg. GET, PUT, POST, DELETE
+   * @param {String} path API route
+   * @param {Object} body Request body
+   */
+  async request(
+    method: HTTPVerb,
+    path: string,
+    body: any,
+    // eslint-disable-next-line unicorn/no-object-as-default-parameter
+    options: {
+      checkCurrentApiIsUp?: boolean;
+      returnOriginalResponse?: boolean;
+    } = { checkCurrentApiIsUp: true, returnOriginalResponse: false },
```

Comment:
Default parameters should not be passed to a function through an object literal. The `foo = {a: false}` parameter works fine if only used with one option. As soon as additional options are added, you risk replacing the whole `foo = {a: false, b: true}` object when passing only one option: `{a: true}`. For this reason, object destructuring should be used instead.

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-object-as-default-parameter.md

Reply:
Fixed